modules/EPMC.py

```diff
@@ -58,8 +51,11 @@ def main(cooccurrenceFile, outputFile, local=False):
         # Reading file:
         spark.read.parquet(cooccurrenceFile)
 
+        # Filter out pairs found in unwanted sections
+        .filter(pf.col('section').isin(SECTIONS_OF_INTEREST))
+
         # Casting integer pmid column to string:
-        .withColumn("pmid", pf.col('pmid').cast(StringType()))
+        .withColumn("pmid", pf.trim(pf.col('pmid').cast(StringType())))
```

Comment:
That's a good question! The story of this `trim()` call is as follows. When I was originally running this, validation of the EPMC results failed because _one record_ contained an extra space symbol in its `pmid` field. This _probably_ (I didn't check, because I really didn't have time in between fixing all of the other bugs :-) caused Spark to either load the entire column as strings, or just that one record.

So, in sort, it's not that `cast(StringType())` does anything nasty, it's just to guard against similar issues in the original data.

Reply:
It's interesting... I vaguely remember adding this statement as the evidence was failing validation as all pmids were integers. I guess at some point the problematic pmid got introduced and since then, spark probably won't cast that column into integer. Good. Thanks for clarifying.